packages/server/lib/plugins/child/run_require_async_child.js

```diff
@@ -78,12 +78,55 @@ function run (ipc, configFile, projectRoot) {
     return false
   }
 
-  ipc.on('loadConfig', () => {
+  // Config file loading of modules is tested within
+  // system-tests/projects/config-cjs-and-esm/*
+  const loadConfig = async (configFile) => {
+    // 1. Try loading the configFile
+    // 2. Catch the "ERR_REQUIRE_ESM" error
+    // 3. Check if esbuild is installed
+    //   3a. Yes: Use bundleRequire
+    //   3b. No: Continue through to `await import(configFile)`
+    // 4. Use node's dynamic import to import the configFile
+
+    try {
+      return require(configFile)
+    } catch (err) {
+      if (!err.stack.includes('[ERR_REQUIRE_ESM]') && !err.stack.includes('SyntaxError: Cannot use import statement outside a module')) {
+        throw err
+      }
+    }
+
+    debug('User is loading an ESM config file')
+
+    try {
+      debug('Trying to use esbuild to run their config file.')
+      // We prefer doing this because it supports TypeScript files
+      require.resolve('esbuild')
+
+      debug(`They have esbuild, so we'll load the configFile via bundleRequire`)
+      const { bundleRequire } = require('bundle-require')
+
+      return (await bundleRequire({ filepath: configFile })).mod
+    } catch (err) {
+      if (err.stack.includes(`Cannot find package 'esbuild'`)) {
+        debug(`User doesn't have esbuild. Going to use native node imports.`)
+
```

Comment:
Do we expect this to work? Should we just throw an error that's like 

"Hey, Cypress depends on `esbuild` for ESM support, please install it and try again"

Reply:
Should we have it as a "real" dependency and only use it if typescript is detected?

Reply:
This works fine for all TS projects where type != module. We don't need to fail -- we can just fall back.